### PR TITLE
users auth

### PR DESCRIPTION
**Pull request by marianahuh**
POST /register and /login works.

### CODE REVIEW

**Comment by marianahuh** [comment]
moved GET /users to correct router

File: auth/authenticate-token.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    username: user.username
+  };
+  const secret = secrets.jwtSecret;
+  const options = {
+    expiresIn: '1hr'
+  };
+  return jwt.sign(payload, secret, options);
+}
+
+function validateToken(req, res, next) {
```
Comment (shobhitav) [inline auth/authenticate-token.js]: Is this the same as authenticate middleware ?

File: auth/authRouter.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      } else {
+        res.status(401).json({ message: 'Invalid credentials' });
+      }
+    })
+    .catch(error => {
+      res.status(500).json(error);
+    });
+});
+
+router.get('/users', (req, res) => {
```
Comment (shobhitav) [inline auth/authRouter.js]: Should not we validate token here

File: auth/authenticate-token.js (hunk 1 above)
Comment (marianahuh) [reply auth/authenticate-token.js]: I separated the middleware to a different file. Token validation is now passing.

File: auth/authRouter.js (hunk 2 above)
Comment (marianahuh) [reply auth/authRouter.js]: Yes. corrected.

**Comment by marianahuh** [comment]
Users validation passes.

**Pull request merged**